lib/internal/repl.js

```diff
@@ -55,10 +55,19 @@ function createRepl(env, opts, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
-    return setupHistory(repl, env.NODE_REPL_HISTORY,
-                        env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal) {
+    // Make sure we trim the history path. We don't want to create
+    // a file name " ".
+    let historyPath = env.NODE_REPL_HISTORY;
```

Comment:
All of this logic, including the check for empty string might make more sense inside of `setupHistory()`.